examples/virtual-machines/linux/load-balanced/main.tf

```
@@ -0,0 +1,131 @@
+  name                = "${var.prefix}-pip"
+  resource_group_name = azurerm_resource_group.main.name
+  location            = azurerm_resource_group.main.location
+  allocation_method   = "Dynamic"
+}
+
+resource "azurerm_network_interface" "main" {
+  count               = local.instance_count
+  name                = "${var.prefix}-nic${count.index}"
+  resource_group_name = azurerm_resource_group.main.name
+  location            = azurerm_resource_group.main.location
+
+  ip_configuration {
+    name                          = "primary"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_availability_set" "avset" {
+  name                         = "${var.prefix}avset"
+  location                     = azurerm_resource_group.main.location
+  resource_group_name          = azurerm_resource_group.main.name
+  platform_fault_domain_count  = 2
+  platform_update_domain_count = 2
+  managed                      = true
+}
+
+resource "azurerm_network_security_group" "webserver" {
+  name                = "tls_webserver"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+  security_rule {
+    access                     = "Allow"
+    direction                  = "Inbound"
+    name                       = "tls"
+    priority                   = 100
+    protocol                   = "Tcp"
+    source_port_range          = "*"
+    source_address_prefix      = "*"
+    destination_port_range     = "443"
+    destination_address_prefix = azurerm_subnet.internal.address_prefix
+  }
+}
+
+resource "azurerm_lb" "example" {
+  name                = "TestLoadBalancer"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  frontend_ip_configuration {
+    name                 = "PublicIPAddress"
+    public_ip_address_id = azurerm_public_ip.pip.id
+  }
+}
+
+resource "azurerm_lb_backend_address_pool" "example" {
+  resource_group_name = azurerm_resource_group.main.name
+  loadbalancer_id     = azurerm_lb.example.id
+  name                = "BackEndAddressPool"
+}
+
+resource "azurerm_lb_nat_rule" "example" {
+  resource_group_name            = azurerm_resource_group.main.name
+  loadbalancer_id                = azurerm_lb.example.id
+  name                           = "HTTPSAccess"
+  protocol                       = "Tcp"
+  frontend_port                  = 443
+  backend_port                   = 443
+  frontend_ip_configuration_name = azurerm_lb.example.frontend_ip_configuration[0].name
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "example" {
+  count                   = local.instance_count
+  backend_address_pool_id = azurerm_lb_backend_address_pool.example.id
+  ip_configuration_name   = "primary"
+  network_interface_id    = azurerm_network_interface.main[count.index].id
```

does this want to be:

```suggestion
  network_interface_id    = element(azurerm_network_interface.main.*.id, count.index)
```